prov/efa/src/efa_msg.c

```diff
@@ -342,7 +342,7 @@ static ssize_t efa_post_send(struct efa_ep *ep, const struct fi_msg *msg, uint64
 
 	efa_post_send_sgl(ep, msg, ewr);
 
-	if (flags & FI_INJECT)
+	if (len <= ep->info->tx_attr->inject_size)
 		wr->send_flags |= IBV_SEND_INLINE;
```

Comment:
The inject size validation should already be handled in the upper layers. My original PR had validation, but dropped it based on Gal's review: https://github.com/ofiwg/libfabric/pull/5573

IIRC, my earlier change was intentional, to not implicitly inline sends unless the app requests it (I lost context though). On completions, you can squash them anyway.

Reply:
I think the main point of this PR is that inline send and inject does not need to be entagled.

As long as the size allows, we should use inline send, because it has performance benefits and does no harm. This is regardless whether INJECT is specified.